File: 342.power-of-four.js
/*
 * @lc app=leetcode id=342 lang=javascript
 *
 * [342] Power of Four
 */

// @lc code=start
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfFour = function(n) {
    if (n === 1) return true
    if (n < 1) return false

    while (n > 1) {
        if (n % 4 === 0) {
            n = n / 4
        } else {
            return false
        }
    }
    return true
};
// @lc code=end

File: 342.power-of-four.py
#
# @lc app=leetcode id=342 lang=python
#
# [342] Power of Four
#

# @lc code=start
class Solution(object):
    def isPowerOfFour(self, n):
        """
        :type n: int
        :rtype: bool
        """

        if n == 0:
            return False

        while n % 4 == 0:
            n /= 4

        return n == 1

# @lc code=end
